### ISSUE TITLE
Numpy 2 compatability

### ISSUE THREAD

**Issue by joeloskarsson** [mllam/neural-lam#67]
Nerual-lam is at the moment not completely compatible with the [new release of numpy 2](https://numpy.org/devdocs/release/2.0.0-notes.html). At the moment the requirements specification includes numpy 2, so that is what will be installed when installing neural-lam. 

While the upgrade to numpy 2 compatability is likely not huge, it is also not urgent. It is still good to first prevent this issue by changing the dependency specifier, and then we can put it on the roadmap to upgrade.

Things TODO:

- [x] Change requirement specifier to prevent installs of numpy 2
- [x] Put upgrade on roadmap
- [ ] Finish numpy 2 compatability upgrade

**Comment by joeloskarsson** [comment]
One error message we get with numpy 2 is "Alias np.float_ has been removed. Use np.float64 instead." (from @sadamov)

**Comment by joeloskarsson** [comment]
@sadamov Do you remember any more information about where you ran into `Alias np.float_ has been removed. Use np.float64 instead.`? I managed to run training and testing successfully on CPU with numpy 2.0.1. Was that error from some of the pre-processing scripts? (I have not tried them yet)

**Comment by sadamov** [comment]
I have run all pre-processing scripts and the model training with the latest env from https://github.com/mllam/neural-lam/pull/32. Tested both on CPU/GPU to be sure, no issues. So this must mean that I was previously working with a dirty environment. Sometimes I managed to install numpy<2.0.0 with mamba and numpy>2.0.0 with pip on top. Probably some libraries in my env were still built against numpy=1.2.6 as a result. I apologies for wasting our time and suggest to close this issue. I will update my comment in https://github.com/mllam/neural-lam/pull/37 and reference this issue.
